core/src/test/scala/com/gu/tableversions/core/TableVersionsSpec.scala

```diff
@@ -187,38 +191,278 @@ trait TableVersionsSpec {
+
+        _ <- tableVersions.commit(table, tableUpdate1)
+        _ <- tableVersions.commit(table, tableUpdate2)
+        _ <- tableVersions.commit(table, tableUpdate3)
+
+        versionAfterWrites <- tableVersions.currentVersion(table)
+
+        // Get the update history after updates
+        fullHistory <- tableVersions.updates(table)
+
+        // Setting the current version to the latest should have no effect
+        _ <- tableVersions.setCurrentVersion(table, fullHistory.head.id)
+        versionAfterWrites2 <- tableVersions.currentVersion(table)
+
+        // Set the initial version, i.e. the table as it was before any committed version
+        _ <- tableVersions.setCurrentVersion(table, fullHistory.last.id)
+        versionSetToInitial <- tableVersions.currentVersion(table)
+
+        // Set the version after the first commit
+        _ <- tableVersions.setCurrentVersion(table, fullHistory.takeRight(2).head.id)
+        versionSetToFirst <- tableVersions.currentVersion(table)
+
+        // Set the second from last version
+        _ <- tableVersions.setCurrentVersion(table, fullHistory.drop(1).head.id)
+        versionSetToSecond <- tableVersions.currentVersion(table)
+
+        // Set the latest version again
+        _ <- tableVersions.setCurrentVersion(table, fullHistory.head.id)
+        versionSetToLatest <- tableVersions.currentVersion(table)
+
+      } yield
+        (historyAfterInit,
+         fullHistory,
+         versionAfterWrites,
+         versionAfterWrites2,
+         versionSetToInitial,
+         versionSetToFirst,
+         versionSetToSecond,
+         versionSetToLatest)
+
+      val (historyAfterInit,
+           fullHistory,
+           versionAfterWrites,
+           versionAfterWrites2,
+           versionSetToInitial,
+           versionSetToFirst,
+           versionSetToSecond,
+           versionSetToLatest) = scenario.unsafeRunSync()
+
+      historyAfterInit should have size 1
+      val initUpdate = historyAfterInit.head
+
+      fullHistory should contain theSameElementsInOrderAs List(tableUpdate3.metadata,
+                                                               tableUpdate2.metadata,
+                                                               tableUpdate1.metadata,
+                                                               initUpdate)
+
+      // Commit IDs should be unique
+      fullHistory.map(_.id).distinct should contain theSameElementsAs fullHistory.map(_.id)
+
+      versionAfterWrites shouldBe SnapshotTableVersion(version3)
+      versionAfterWrites2 shouldEqual versionAfterWrites
+
+      versionSetToInitial shouldBe SnapshotTableVersion(Version.Unversioned)
+      versionSetToFirst shouldBe SnapshotTableVersion(version1)
+      versionSetToSecond shouldBe SnapshotTableVersion(version2)
+      versionSetToLatest shouldEqual versionAfterWrites
+    }
+
+    it should "return updates in the same order as they were committed, but with the most recent first" in {
```

Comment:
Now with added tests for the ordering of the update history returned by the implementation.